File: src/main/java/com/rover12421/csdn/java2kt/java/_2_NumberAutoUpType.java
package com.rover12421.csdn.java2kt.java;

/**
 * Created by rover12421 on 8/10/16.
 */
public class _2_NumberAutoUpType {
    public static void main(String[] args) {
        int a = 100;
        short b = 2;
        char c = 'a';
        long l = 99;

        System.out.println("a | b = " + (a | b));
        System.out.println("b + c = " + (b + c));
        System.out.println("a & c = " + (a & c));
        System.out.println("a ^ l = " + (a ^ l));
    }
}

File: src/main/kotlin/com/rover12421/csdn/java2kt/kt/_2_NumberAutoUpType.kt
package com.rover12421.csdn.java2kt.kt

/**
 * Created by rover12421 on 8/10/16.
 */
object _2_NumberAutoUpType {
    @JvmStatic fun main(args: Array<String>) {
        val a = 100
        val b: Short = 2
        val c = 'a'
        val l: Long = 99

        println("a | b = " + (a or b.toInt()))
        println("b + c = " + (b + c.toShort()))
        println("a & c = " + (a and c.toInt()))
        println("a ^ l = " + (a.toLong() xor l))
    }

}
